udata_hydra/cli.py

```diff
@@ -291,16 +291,32 @@ async def purge_csv_tables(quiet: bool = False):
     WHERE checks.parsing_table IS NOT NULL AND (c.id IS NULL OR c.deleted = TRUE);
     """
     conn = await connection()
-    tables_to_delete = await conn.fetch(q)
-    tables_to_delete = [table["parsing_table"] for table in tables_to_delete]
+    res: list[Record] = await conn.fetch(q)
+    tables_to_delete: list[str] = [r["parsing_table"] for r in res]
+
+    success_count = 0
+    error_count = 0
 
     for table in tables_to_delete:
-        log.debug(f"Deleting table {table}")
-        await delete_table(table)
-        await conn.execute("UPDATE checks SET parsing_table = NULL WHERE parsing_table = $1", table)
-    if len(tables_to_delete):
-        log.info(f"Deleted {len(tables_to_delete)} table(s).")
-    else:
+        try:
+            async with conn.transaction():
+                log.debug(f'Deleting table "{table}"')
+                await conn.execute(f'DROP TABLE IF EXISTS "{table}"')
+                await conn.execute("DELETE FROM tables_index WHERE parsing_table = $1", table)
+                await conn.execute(
+                    "UPDATE checks SET parsing_table = NULL WHERE parsing_table = $1", table
```

Comment:
Could we add a foreign key constraints as was done in https://github.com/datagouv/hydra/pull/195/, but to a table name?

Reply:
Good idea, I suggest we do that later in another PR, after stabilising the current hydra and deploying the new features, though.
Here is the dedicated issue: https://github.com/datagouv/hydra/issues/231